ops/jenkins/bfd-deploy-apps/Jenkinsfile

```
@@ -0,0 +1,340 @@
+    if (!buildStatus) {
+        buildStatus = 'SUCCESS'
+    }
+
+    // build colors
+    def colorMap = [:]
+    colorMap['STARTED']  = '#0000FF'
+    colorMap['SUCCESS']  = '#00FF00'
+    colorMap['ABORTED']  = '#6A0DAD'
+    colorMap['UNSTABLE'] = '#FFFF00'
+    colorMap['FAILED']   = '#FF0000'
+    def buildColor = colorMap[buildStatus]
+    buildColor = buildColor ?: '#FF0000' // default to red
+
+    def branchUrl = "https://github.com/CMSgov/beneficiary-fhir-data/tree/${env.BRANCH_NAME}"
+
+    def msg = ''
+    switch (buildStatus) {
+        case 'UNSTABLE':
+        case 'SUCCESS':
+            msg = 'COMPLETED SUCCESSFULLY: DEPLOYED BFD APPS AND AMIS'
+            break
+        case 'FAILED':
+        case 'FAILURE':
+            msg = "FAILED ON ${stageName.toUpperCase()} STAGE ON BRANCH <${branchUrl}|${env.BRANCH_NAME.toUpperCase()}>"
+            break
+        case 'STARTED':
+            msg = 'HAS STARTED'
+            break
+        case 'ABORTED':
+            msg = 'WAS ABORTED'
+            break
+        default:
+            msg = "${buildStatus.toUpperCase()}"
+            break
+    }
+
+    // build slack message
+    def slackMsg = ''
+    if (buildStatus == 'UNSTABLE') {
+        slackMsg = "UNSTABLE BFD APP/AMI DEPLOY <${env.BUILD_URL}|#${env.BUILD_NUMBER}> ON BRANCH <${branchUrl}|${env.BRANCH_NAME.toUpperCase()}> ${msg} \n"
+    } else {
+        slackMsg = "BFD APP/AMI DEPLOY <${env.BUILD_URL}|#${env.BUILD_NUMBER}> ON BRANCH <${branchUrl}|${env.BRANCH_NAME.toUpperCase()}> FOR ${environment.toUpperCase()} ENVIRONMENT ${msg} \n"
+    }
+    slackMsg += "\tJob '${env.JOB_NAME}' \n"
+
+    // send Slack messages
+    slackSend(color: buildColor, message: slackMsg)
+}
+
+pipeline {
+    agent {
+        kubernetes {
+            defaultContainer 'bfd-cbc-build'
+            yaml """
+apiVersion: v1
+kind: Pod
+spec:
+  serviceAccount: bfd
+  containers:
+  - name: bfd-cbc-build
+    image: 'public.ecr.aws/c2o1d8s9/bfd-cbc-build:jdk17-mvn3-tfenv3-latest'
+    command:
+    - cat
+    tty: true
+    imagePullPolicy: Always
+    resources:
+      requests:
+        memory: '16384Mi'
+        cpu: '8000m'
+      limits:
+        memory: '16384Mi'
+        cpu: '8000m'
+"""
+        }
+    }
+
+    parameters {
+        string(
+        name: 'env',
+        description: 'The BFD Environment to deploy apps to'
+        )
+
+        string(
+        name: 'migrator_ami_override',
+        description: 'The AMI ID to deploy migrator from',
+        defaultValue: null
+        )
+
+        string(
+        name: 'pipeline_ami_override',
+        description: 'The AMI ID to deploy pipeline from',
+        defaultValue: null
+        )
+
+        string(
+        name: 'server_ami',
+        description: 'The AMI ID to deploy server from',
+        defaultValue: null
+        )
+
+        booleanParam(
+            name: 'force_migrator_deployment', 
+            description: 'When true, force the migrator to deploy.', 
```

```suggestion
            description: 'When true, force the migrator to deploy', 
```

(no period, or add periods to the other descriptions) 😉